app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -171,57 +129,190 @@ fun OverviewScreen(
-                    trips.sortedBy { it.startDate }
-                  } else {
-                    trips
-                        .filter { it.name.contains(searchQuery, ignoreCase = true) }
-                        .sortedBy { it.startDate }
-                  }
+          Toast.makeText(context, R.string.notification_no_internet_text, Toast.LENGTH_SHORT).show()
+        }
+      },
+      modifier = Modifier.testTag("createTripButton")) {
+        Icon(
+            Icons.Outlined.Add,
+            stringResource(R.string.floating_button),
+            modifier = Modifier.testTag("addIcon"))
+      }
+}
 
-              // Add this condition to check if filteredTrips is empty while searching
-              if (searchQuery.isNotEmpty() && filteredTrips.isEmpty()) {
-                Box(modifier = Modifier.fillMaxSize()) {
-                  NoResultsFound(modifier = Modifier.testTag("noSearchResults"))
-                }
-              } else {
-                LazyColumn(
-                    verticalArrangement = Arrangement.spacedBy(10.dp, Alignment.Top),
-                    horizontalAlignment = Alignment.CenterHorizontally,
-                    modifier = Modifier.fillMaxSize().testTag("lazyColumn")) {
-                      filteredTrips.forEach { trip ->
-                        item {
-                          TripItem(
-                              tripsViewModel = tripsViewModel,
-                              trip = trip,
-                              navigationActions = navigationActions,
-                              userViewModel = userViewModel)
-                          Spacer(modifier = Modifier.height(10.dp))
-                        }
-                      }
-                    }
-              }
-            }
+/**
+ * Composable that renders the main content of the overview screen.
+ *
+ * Handles the display of loading indicator, empty state, and the list of trips based on the current
+ * state of the app.
+ *
+ * @param isLoading Boolean indicating whether data is currently being loaded
+ * @param trips List of all available trips
+ * @param searchQuery Current search query for filtering trips
+ * @param padding PaddingValues to apply to the content
+ * @param tripsViewModel ViewModel containing trips-related state and logic
+ * @param navigationActions Actions to handle navigation between screens
+ * @param userViewModel ViewModel containing user-related state and logic
+ */
+@Composable
+private fun OverviewContent(
+    isLoading: Boolean,
+    trips: List<Trip>,
+    searchQuery: String,
+    padding: PaddingValues,
+    tripsViewModel: TripsViewModel,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+  if (isLoading) {
+    CircularProgressIndicator(modifier = Modifier.testTag("loadingIndicator"))
+    return
+  }
+
+  Column(
+      modifier = Modifier.padding(padding).testTag("overviewColumn"),
+  ) {
+    when {
+      trips.isEmpty() -> EmptyTripsMessage()
+      else ->
```

Comment:
This can be simplified to an if / else as we only have 2 cases

Reply:
Thank you for this pointing this out! I overlooked the code and did not notice